addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/internal/device/DSMRSerialAutoDevice.java

```diff
@@ -0,0 +1,276 @@
+     */
+    private @Nullable ScheduledFuture<?> endTimeTimer;
+
+    /**
+     * The listener of the class handling the connector events
+     */
+    private DSMREventListener parentListener;
+
+    /**
+     * Time in nanos the last time the baudrate was switched. This is used during discovery ignore errors retrieved
+     * after switching baudrate for the period set in {@link #SWITCHING_BAUDRATE_TIMEOUT_NANOS}.
+     */
+    private long lastSwitchedBaudrateNanos;
+
+    /**
+     * Creates a new {@link DSMRSerialAutoDevice}
+     *
+     * @param serialPortManager the manager to get a new serial port connecting from
+     * @param serialPortName the port name (e.g. /dev/ttyUSB0 or COM1)
+     * @param listener the parent {@link DSMREventListener}
+     * @param scheduler the scheduler to use with the baudrate switching timers
+     * @param baudrateSwitchTimeoutSeconds timeout period for when to try other baudrate settings and end the discovery
+     *            of the baudrate
+     */
+    public DSMRSerialAutoDevice(SerialPortManager serialPortManager, String serialPortName, DSMREventListener listener,
+            ScheduledExecutorService scheduler, int baudrateSwitchTimeoutSeconds) {
+        this.parentListener = listener;
+        this.scheduler = scheduler;
+        this.baudrateSwitchTimeoutSeconds = baudrateSwitchTimeoutSeconds;
+        telegramListener = new DSMRTelegramListener(listener);
+        dsmrConnector = new DSMRSerialConnector(serialPortManager, serialPortName, telegramListener);
+        logger.debug("Initialized port '{}'", serialPortName);
+    }
+
+    @Override
+    public void start() {
+        stopDiscover(DeviceState.DISCOVER_SETTINGS);
+        portSettings = DEFAULT_PORT_SETTINGS;
+        telegramListener.setDsmrEventListener(this);
+        dsmrConnector.open(portSettings);
+        restartHalfTimer();
+        endTimeTimer = scheduler.schedule(this::endTimeScheduledCall,
+                baudrateSwitchTimeoutSeconds * DISCOVER_TIMEOUT_FACTOR, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void restart() {
+        if (state == DeviceState.ERROR) {
+            stop();
+            start();
+        } else if (state == DeviceState.NORMAL) {
+            dsmrConnector.restart(portSettings);
+        }
+    }
+
+    @Override
+    public synchronized void stop() {
+        dsmrConnector.close();
+        stopDiscover(state);
+        logger.trace("stopped with state:{}", state);
+    }
+
+    /**
+     * Handle if telegrams are received.
+     *
+     * @param telegram the details of the received telegram
+     */
+    @Override
+    public void handleTelegramReceived(P1Telegram telegram) {
+        if (!telegram.getCosemObjects().isEmpty()) {
+            stopDiscover(DeviceState.NORMAL);
+            parentListener.handleTelegramReceived(telegram);
+            logger.info("Start receiving telegrams on port {} with settings: {}", dsmrConnector.getPortName(),
```

Comment:
You can use "info" if this is a one time message at connection time, but here it seems that it can happen pretty much of the time?